packages/flutter_tools/lib/src/flx.dart

```diff
@@ -36,7 +36,7 @@ Future<int> createSnapshot({
   assert(mainPath != null);
   assert(snapshotPath != null);
   assert(packages != null);
-  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot);
+  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot, null, BuildMode.debug);
```

Comment:
gen_snapshot will be found at different paths in the local engine output depending on which build mode the engine is targeting.  A script snapshot implies debug mode, so this passes BuildMode.debug